js/modules.js

```diff
@@ -15,52 +15,79 @@ CORE.create_module("email-nav", function (sb) {
 CORE.create_module("email-catalog", function (sb) {
     var emails;
 
+    var openEmail = function(e) {
+        sb.hide();
+        sb.notify({
+            type : "open-email",
+            data : e.currentTarget.getAttribute("data-email-id")
```

Comment:
I think thats what worries me the most. If i somehow managed to get an id for an email that was not sent to me, i could insert it into the DOM and read it. If we can't find a solution to putting IDs in the DOM, then we should add a security check anytime you try to retrieve an email that check if you have access to it. Actually we should be doing that regardless. I'll make an issue for it